admin/controllers/service-manager.js

```diff
@@ -13,374 +13,343 @@ const sortingAttributesService = require('../services/sorting-attributes.service
+        action: urlActionName,
+        checkWindowData: req.body,
+        error: validationError.errors,
+        errorMessage: checkWindowErrorMessages,
+        currentYear,
+        actionName,
+        urlActionName,
+        breadcrumbs: req.breadcrumbs(),
+        adminIsDisabled,
+        checkStartIsDisabled
+      })
+    }
 
-  try {
-    if (insertNew) {
-      await checkWindowDataService.sqlCreate(checkWindow)
-    } else {
-      await checkWindowDataService.sqlUpdate(checkWindow)
+    try {
+      if (req.body.checkWindowId) {
+        checkWindow = await checkWindowDataService.sqlFindOneById(req.body.checkWindowId)
+      }
+    } catch (error) {
+      return next(error)
     }
-    req.flash('info', flashMessage)
-  } catch (error) {
-    winston.error('Could not save check windows data.', error)
-    return next(error)
-  }
 
-  return res.redirect('/service-manager/check-windows')
-}
+    let insertNew = false
 
-/**
- * Soft delete a check window by id.
- * @param req
- * @param res
- * @param next
- * @returns {Promise.<*>}
- */
-const removeCheckWindow = async (req, res, next) => {
-  let checkWindow
-
-  if (!req.params.checkWindowId) {
-    return res.redirect('/service-manager/check-windows')
-  }
+    if (typeof checkWindow === 'undefined') {
+      insertNew = true
+      checkWindow = {}
+    }
 
-  try {
-    checkWindow = await checkWindowDataService.sqlFindOneById(req.params.checkWindowId)
-  } catch (err) {
-    return next(err)
-  }
+    checkWindow.name = req.body[ 'checkWindowName' ]
+    if (req.body[ 'adminStartDay' ] && req.body[ 'adminStartMonth' ] && req.body[ 'adminStartYear' ]) {
+      checkWindow.adminStartDate = dateService.createLocalTimeFromDayMonthYear(req.body[ 'adminStartDay' ], req.body[ 'adminStartMonth' ], req.body[ 'adminStartYear' ])
+    }
+    if (req.body[ 'checkStartDay' ] && req.body[ 'checkStartMonth' ] && req.body[ 'checkStartYear' ]) {
+      checkWindow.checkStartDate = dateService.createLocalTimeFromDayMonthYear(req.body[ 'checkStartDay' ], req.body[ 'checkStartMonth' ], req.body[ 'checkStartYear' ])
+    }
+    checkWindow.checkEndDate = dateService.createLocalTimeFromDayMonthYear(req.body[ 'checkEndDay' ], req.body[ 'checkEndMonth' ], req.body[ 'checkEndYear' ])
+    // Ensure check end date time is set to the last minute of the particular day
+    checkWindow.checkEndDate.set({ hour: 23, minute: 59, second: 59 })
 
-  if (checkWindow) {
-    if (Date.parse(checkWindow.checkStartDate) < moment.now()) {
-      req.flash('error', 'Deleting an active check window is not allowed.')
-    } else {
-      try {
-        await checkWindowDataService.sqlDeleteCheckWindow(req.params.checkWindowId)
-        req.flash('info', 'Check window deleted.')
-      } catch (error) {
-        req.flash('error', 'Error trying to delete check window.')
+    // Auditing? Question for BAs.
+
+    try {
+      if (insertNew) {
+        await checkWindowDataService.sqlCreate(checkWindow)
```

Comment:
can you push this data service call down into a service?

Reply:
@GuyHarwood this part appears as changed because I have changed the controller declaration but these changes are going to be tackled when i start refactoring the save check window related part.